### PR TITLE
feat: Support for vSAN Stretched Clusters

### PR DESCRIPTION
**Pull request by stoyanzhelyazkov**
**Summary of Pull Request**

Adding a new capability to the cluster resource that enables vSAN to be configured in stretched mode.

The new capability is controlled via a new input property called "stretch_configuration".
Inside one can configure both the witness host and the additional hosts that will be commissioned into the cluster.

Removing the new property from a cluster's configuration signals that the cluster should go back into single-site mode (unstretched)

It is not possible to create a stretched cluster via the SDDC manager and this operation is explicitly forbidden in the provider.
Users will be advised to first create their cluster and apply the advanced vSAN configuration in a subsequent update.

**Type of Pull Request**

- [ ] This is a bug fix.
- [X] This is an enhancement or feature.
- [ ] This is a code style/formatting update.
- [ ] This is a documentation update.
- [ ] This is a refactoring update.
- [ ] This is a chore update
- [ ] This is something else.
      Please describe:

**Related to Existing Issues**

Issue Number: N/A

**Test and Documentation Coverage**

For bug fixes or features:

- [X] Tests have been completed.
- [X] Documentation has been added/updated.

**Breaking Changes?**

- [ ] Yes, there are breaking changes.
- [X] No, there are no breaking changes.

### CODE REVIEW

File: internal/provider/resource_cluster_test.go [hunk 1]
```diff
@@ -52,6 +52,43 @@ func TestAccResourceVcfClusterCreate(t *testing.T) {
...
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testCheckVcfClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccVcfClusterResourcеStretchTestConfig(
+					"sfo-w02-cl02",
+					""),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("vcf_cluster.cluster", "is_stretched"),
```
Comment (dimitarproynov) [inline internal/provider/resource_cluster_test.go:66]: Maybe it is worth checking if the value is "false", then "true", then "false" again.

**Review by dimitarproynov — DISMISSED** [review]
Stellar!

File: internal/provider/resource_cluster_test.go (hunk 1 above)
Comment (stoyanzhelyazkov) [reply internal/provider/resource_cluster_test.go]: Good idea, done

**Review by dimitarproynov — DISMISSED** [review]

**Comment by stoyanzhelyazkov** [comment]
@dimitarproynov I renamed "stretch_configuration" to "vsan_stretch_configuration"
Which means I have to ask you to stamp this PR one more time

**Review by dimitarproynov — APPROVED** [review]

**Pull request merged**